proposal.rst

```diff
@@ -150,43 +197,134 @@ Motivations
+in research papers by giving equation as query. Currently he/she has to put the math 
+formula name or search for concept name.
 
 Project Details
 ---------------
 
-.. Please go into plenty of detail in this section.
+.. Please go into plenty of detail in thi-s section.
 
 **Describe any existing work and concepts on which your project is based.**
 
-FILLME
+The system I propose is based on Tangent - math expression search engine created 
+by David Stalnaker and it's improved versions developed at `dprl, RIT 
+<https://www.cs.rit.edu/~dprl/Software.html#tangent-s>`_. Tangent performs indexing
+on the structure of the expression represented in MathML format. Encoding of structure 
+is done by constructing symbol layout tree from the expression and then generating 
+symbol pairs from that tree. The symbol pairs are indexed using inverted index. 
 
 **Do you have any preliminary findings or results which suggest that your
 approach is possible and likely to succeed?**
 
-FILLME
+Tangent system code is available for public. I successfully ran the system and 
+reproduced some of the results. Also, team who developed Tangent participated in `NTCIR
+math retrieval task <http://ntcir-math.nii.ac.jp/>`_ and obtained competitive results 
+for arXiv-main task [1]_.
 
 **What other approaches to have your considered, and why did you reject those in
 favour of your chosen approach?**
 
-FILLME
+I considered MIaS system [2]_, which linearizes math expression into text and uses 
+traditional text retrieval search engine. Indexing part is similar to the Tangent 
+system, approach I would like to propose. It differs in the preprocessing of math
+expression to generate indexes. MIaS uses tokenization of expression and performs
+unification [*]_ strategy to generalize the expression. The reason for selecting 
+the approach used in Tangent system is, symbol layout tree approach extracts the 
+structural information in expression better than tokenization of text. This is 
+corroborated by the findings of hypothesis test. Also, Tangent system performance
+is better than MIaS system.
+
+I also studied the system [3]_ implemented by MCAT group. They include textual context
+of the formulae and integrate retrieval of text and formulae. The system extracts 
+three granularity level text information. Even though this system performs better
+than Tangent, due to the complexity and keeping time constraint in mind, I planned
+to pick Tangent system. This could be tried in Xapian in future.
 
 **Please note any uncertainties or aspects which depend on further research or
 investigation.**
 
-FILLME
+ We have decided to support latex format for documents and queries. This requires
+ conversion of latex format to mathml format. As this function is in itself can be 
+ a project, we plan to use third party tool like latexmathml. But the exact tool 
+ choice is not finalised yet. This feature will be floater in the work pipeline.
+
```

Comment:
Why is there a need to convert to mathml here at all?

My understanding of MathML is that it's an XML format for representing equations.

We presumably actually want some sort of in-memory parse tree of the equation - what's the reason behind your choice of getting that from LaTeX by converting to MathML and then parsing that, rather than just by parsing the LaTeX?

Reply:
This is the approach taken in the paper. I didn't think it the way you pointed out.

But this requires writing two types of parser right?

Reply:
It would need a parser for LaTeX and one for MathML to support both, but that's probably easier than writing a LaTeX to MathML convertor and a MathML parser - the problem with writing convertors is that you have to handle any conceptual mismatches between the formats (or lose information, e.g. by throwing away things that don't nicely fit into the target format).

The advantage of converting is you can potentially off-load the work to an existing convertor, though I worry you'll hit the problem of that not converting everything faithfully.  So perhaps it would be better to pick a format to support and worry about other formats later anyway.

But I'm not familiar with MathML and perhaps it's actually just LaTeX recast as XML and so trivial to convert.

I've not had a chance to read the references yet, so not sure about the time for the preprocessing step.  It seems plausible without really knowing what's involved though.

Reply:
I guess latex to mathml conversion done because mathml has various tags(variables, constants, operators) that makes representing different token types easy in a tree.

Reply:
"The advantage of converting is you can potentially off-load the work to an existing convertor, though I worry you'll hit the problem of that not converting everything faithfully."
Right. I remember reading about this point some where in tangent code.